alf/algorithms/algorithm.py

```diff
@@ -1259,12 +1259,13 @@ def _make_time_major(nest):
 
         for u in range(num_updates):
             if mini_batch_size < batch_size:
-                # here we use numpy random.permutation to generate the permuted
-                # indices, as the cuda version of torch.randperm(n) seems to
-                # have a bug when n is a large number, generating negative or
-                # very large values that cause out of bound kernel error
-                # https://github.com/pytorch/vision/issues/3816
-                indices = torch.as_tensor(np.random.permutation(batch_size))
+                # here we use the cpu version of torch.randperm(n) to generate
+                # the permuted indices, as the cuda version of torch.randperm(n)
+                # seems to have a bug when n is a large number, generating
+                # negative or very large values that cause out of bound kernel
+                # error: https://github.com/pytorch/pytorch/issues/59756
+                indices = torch.as_tensor(
+                    torch.randperm(batch_size, device='cpu'))
```

Comment:
as_tensor will not convert it to cuda tensor. You can use `alf.nest.utils.convert_device()` to do the job.

Reply:
Good point. Switched to alf.nest.utils.convert_device().